plugins/templates/removeVersion.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const boom = require('@hapi/boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const nameSchema = schema.models.template.base.extract('name');
+const versionSchema = schema.models.template.base.extract('version');
+
+module.exports = () => ({
+    method: 'DELETE',
+    path: '/templates/{name}/versions/{version}',
+    options: {
+        description: 'Delete the specified version of template and the tags associated with it',
+        notes: 'Returns null if successful',
+        tags: ['api', 'templates'],
+        auth: {
+            strategies: ['token'],
+            scope: ['build', 'user', '!guest']
+        },
+
+        handler: async (request, h) => {
+            const { name, version } = request.params;
+            const { credentials } = request.auth;
+            const { templateFactory, templateTagFactory } = request.server.app;
+            const { canRemove } = request.server.plugins.templates;
+            let shouldUpdateLatest = false;
+
+            return Promise.all([
+                templateFactory.get({ name, version }),
+                templateTagFactory.list({ params: { name, version } })
+            ])
+                .then(([template, tags]) => {
+                    if (!template) {
+                        throw boom.notFound(`Template ${name} with version ${version} does not exist`);
+                    }
+
+                    return canRemove(credentials, template, 'admin', request.server.app)
+                        .then(() => {
+                            shouldUpdateLatest = template.latest;
+                            const templatePromises = [template.remove()];
+                            const tagPromises = tags.map(tag => tag.remove());
+
+                            return Promise.all(templatePromises.concat(tagPromises)).then(() => {
+                                if (shouldUpdateLatest) {
+                                    return templateFactory
+                                        .list({ params: { name }, sort: 'descending', sortBy: 'createTime' })
```

Comment:
I would say, just fetch the first one in descending order? 

https://github.com/screwdriver-cd/models/blob/master/lib/baseFactory.js#L184

from: 
```
.list({ params: { name }, sort: 'descending', sortBy: 'createTime' })
```

to
```
.list({ params: { name }, sort: 'descending', sortBy: 'createTime', paginate: { count: 1 }})
```

Reply:
Done